realm/src/androidTest/java/io/realm/DynamicRealmObjectTest.java

```diff
@@ -0,0 +1,334 @@
+package io.realm;/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import android.test.AndroidTestCase;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmConfiguration;
+import io.realm.dynamic.DynamicRealmList;
+import io.realm.dynamic.DynamicRealmObject;
+import io.realm.entities.AllJavaTypes;
+
+import static io.realm.internal.test.ExtraTests.assertArrayEquals;
+
+public class DynamicRealmObjectTest extends AndroidTestCase {
+
+    private Realm realm;
+    private DynamicRealmObject dObj;
+
+    @Override
+    protected void setUp() throws Exception {
+        RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext()).schema(AllJavaTypes.class).build();
+        Realm.deleteRealm(realmConfig);
+        realm = Realm.getInstance(realmConfig);
+        realm.beginTransaction();
+        AllJavaTypes obj = realm.createObject(AllJavaTypes.class);
+        obj.setColumnString("str");
+        obj.setColumnShort((short) 1);
+        obj.setColumnInt(1);
+        obj.setColumnLong(1);
+        obj.setColumnFloat(1.23f);
+        obj.setColumnDouble(1.234d);
+        obj.setColumnBinary(new byte[]{1, 2, 3});
+        obj.setColumnBoolean(true);
+        obj.setColumnDate(new Date(1000));
+        obj.setColumnObject(obj);
+        obj.getColumnList().add(obj);
+        dObj = new DynamicRealmObject(realm, obj.row);
+        realm.commitTransaction();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
+        realm.close();
+    }
+
+    // Test invalid input: <empty, non-existing field, wrong field type>
+    public void testGetBooleanIllegalArguments() {
+        List<String> arguments = Arrays.asList(null, "foo", "columnString");
+        for (String argument : arguments) {
+            try {
+                dObj.getBoolean(argument);
+                fail(argument + " should throw exception.");
+            } catch (IllegalArgumentException expected) {
+            }
+        }
+    }
+
+    public void testGetBoolean() {
```

Comment:
All these tests are so repetitive but dispersed. I would suggest to make a single testcase testing all GetXX, a single test case testing GetLinkedXX() and a third testing GetXXIllegalArguments.
That would be so much easier to get an overview of.